src/kubernetes/handle_watch_stream.rs

```diff
@@ -0,0 +1,46 @@
+use futures::StreamExt;
+use futures_util::Stream;
+use kube::{
+    runtime::{reflector::store, watcher},
+    Resource,
+};
+use std::{hash::Hash, time::Duration};
+use tokio::pin;
+use tokio_util::time::DelayQueue;
+
+pub async fn handle_watch_stream<K, W>(
```

Comment:
perhaps this should just be named `reflector` since it's _our_ custom `reflector`